### PR TITLE
[JENKINS-25089] Add Auto-Submitted header to emails (RFC 3834)

### PR DESCRIPTION
**Pull request by cpfeiffer**
(Re-add this commit that was originally added as
11142e5f43e55d5bb0f5dd2c8297e737cf122307)

### CODE REVIEW

**Comment by alecharp** [comment]
I know it has been a long time since you open this. I'm sorry for that.

From the RFC you link

> The Auto-Submitted field, with a value of "auto-replied", SHOULD be
included in the message header of any automatic response.  See
section 5.

I understand that it's the message of the automatic response that should have this header.
However, Jenkins emails are not automatic responses but can be seen as "conversation starters".

I'm not sure we want this header in the emails sent by Jenkins.

**Comment by cpfeiffer** [comment]
Maybe this introductory sentence that you quoted is not quite as exact as it should. If you go on to section 5, it reads a little different:
> 5.  The Auto-Submitted header field
>
>   The purpose of the Auto-Submitted header field is to indicate that
   the message was originated by an automatic process, or an automatic
   responder, rather than by a human; and to facilitate automatic
   filtering of messages from signal paths for which automatically
   generated messages and automatic responses are not desirable.

...

> The auto-generated keyword:
>
>   -  SHOULD be used on messages generated by automatic (often periodic)
      processes (such as UNIX "cron jobs") which are not direct
      responses to other messages,

In short: it should be used for *messages*, not just *responses*. The point is to avoid auto-responses like out-of-office reminders.

**Comment by alecharp** [comment]
ok I see, makes more sense then.

However, I can see that was added then removed. Let me find why it was removed.

**Comment by campi01** [comment]
Could this be a configurable option in Jenkins ? This change breaks email sending to ie. sympa mailing lists, which has an explicit regex to catch this "Auto-submitted" header and discard the messages to prevent mail loops.

Also the RFC term "SHOULD" is defined (https://tools.ietf.org/html/rfc2119) as: 

3. SHOULD   This word, or the adjective "RECOMMENDED", mean that there
   may exist valid reasons in particular circumstances to ignore a
   particular item, but the full implications must be understood and
   carefully weighed before choosing a different course.
